components/server/src/user/user-deletion-service.ts

```diff
@@ -50,7 +52,8 @@ export class UserDeletionService {
             log.debug({ userId: id }, "Is deleted but markDeleted already set. Continuing.");
         }
 
-        if (this.config.enablePayment) {
+        const billingMode = await this.billingMode.getBillingModeForUser(user, new Date());
```

Comment:
Let's move that into `await this.stripeService.cancelSubscription(subscriptionId);` as implemented on the usage component. It should be authoritative over the deletion and hence the check should be there such that all invocations of it re-use it

Reply:
Done

Reply:
https://github.com/gitpod-io/gitpod/pull/17318/commits/e7b0e8bc406636e7e7b08af05ab4d3e9224ae506
Maybe there was a GitHub delay again? Workspace showed "push" well before I added the comment :shrug: